src/main/java/com/yahoo/identity/services/credential/CredentialImpl.java

```diff
@@ -14,6 +17,7 @@
 
 public class CredentialImpl implements Credential {
 
+    private final KeyService keyService = new KeyServiceImpl();
```

Comment:
Dependency should be injected.

Reply:
Now all service classes imported in another class are annotated with `@Inject`.

Reply:
Yeah. What you are doing is correct. Dependencies can be injected in many ways, constructor injection, fields injection, setter injection. Constructor injection is the best in most cases I think. 
The annotation you mentioned is used by some injection framework, e.g. Guice. Since we aren't using injection framework, we don't need to add that annotation.